### ISSUE TITLE
The message `This condition will always return true since the function is always defined` is not shown

### ISSUE THREAD

**Issue by a-tarasyuk** [microsoft/TypeScript#37149]
**TypeScript Version:**  3.8.3

**Code**

```ts
const foo = {
    test() {
        return true;
    }
}

if (foo.test) {
    console.log(true);
}

```

**Expected behavior:**
The message `This condition will always return true since the function is always defined` should be shown for the code `if (foo.test)` 

**Actual behavior:**
no message

**Playground Link:** [Example](https://www.typescriptlang.org/play/index.html?module=0#code/MYewdgzgLgBAbgQwDYwLwwN4Cga5gSwgBUAnAVwFMAKASkxhIqjJLBinIoG4YBfLflnwAzGFURIAdIVKU62PDFCQQSCpKQgA5lQ5yuArEA)

It seems that happens when `childSymbol.id` and `testedFunctionSymbol.id` equal to `undefined`.
```ts
if (childSymbol && childSymbol.id === testedFunctionSymbol.id) 
```

**Comment by j-oliveras** [comment]
Using the playground link, after any change (for example add a space at start) the error appears.

**Comment by j-oliveras** [comment]
Probably you already knows, but its a duplicate of #33792.

**Comment by sandersn** [comment]
Yep, thanks @j-oliveras!

**Comment by a-tarasyuk** [comment]
@sandersn I think it is similar to this issue https://github.com/microsoft/TypeScript/issues/35557, however, it is not related to QF.

**Comment by sandersn** [comment]
Maybe, but this issue is fixed in master, whereas #35557 is not. I assumed incorrectly that it was a result of #33792, but I guess not.
